pallets/ddc-verification/src/lib.rs

```diff
@@ -304,24 +336,100 @@ pub mod pallet {
 				"Error retrieving customers activities to validate"
 			);
 			let min_nodes = dac_nodes.len().ilog2() as usize;
-			let _ = Self::get_consensus_for_activities(
+
+			let _customers_activity_mmr_root = match Self::get_consensus_for_activities(
 				&cluster_id,
 				id,
 				&customers_usage,
 				min_nodes,
 				Percent::from_percent(T::MAJORITY),
-			);
-			let _ = Self::get_consensus_for_activities(
+			) {
+				Ok(customers_activity_in_consensus) => {
+					// Process node_activities
+					let mut sorted_activities = customers_activity_in_consensus.clone();
+					sorted_activities.sort();
+
+					let leaves: Vec<ActivityHash> =
+						sorted_activities.iter().map(|activity| activity.hash::<T>()).collect();
+
+					Some(Self::create_merkle_root(leaves))
+				},
+				Err(errors) => {
+					let results = signer.send_signed_transaction(|_account| {
+						Call::emit_consensus_errors { errors: errors.clone() }
+					});
+
+					for (acc, res) in results {
+						match res {
+							Ok(_) => log::info!(
+								"Successfully submitted error processing tx: {:?}",
+								acc.id
+							),
+							Err(e) => log::error!(
+								"Failed to submit error processing tx: {:?} ({:?})",
+								acc.id,
+								e
+							),
+						}
+					}
+					None
+				},
+			};
+
+			let _nodes_activity_mmr_root = match Self::get_consensus_for_activities(
 				&cluster_id,
 				id,
 				&nodes_usage,
 				min_nodes,
 				Percent::from_percent(T::MAJORITY),
-			);
+			) {
+				Ok(nodes_activity_in_consensus) => {
+					// Process node_activities
+					let mut sorted_activities = nodes_activity_in_consensus.clone();
+					sorted_activities.sort();
+
+					let leaves: Vec<ActivityHash> =
+						sorted_activities.iter().map(|activity| activity.hash::<T>()).collect();
```

Comment:
maybe worthwhile to use trie. ordered_trie_root and remove this sorting

Reply:
Done